subsys/logging/log_backend_uart.c

```diff
@@ -10,23 +10,38 @@
 #include <logging/log_output.h>
 #include <logging/log_output_dict.h>
 #include <logging/log_backend_std.h>
+#include <logging/log.h>
 #include <device.h>
 #include <drivers/uart.h>
 #include <sys/__assert.h>
+LOG_MODULE_REGISTER(log_uart);
 
 /* Fixed size to avoid auto-added trailing '\0'.
  * Used if CONFIG_LOG_BACKEND_UART_OUTPUT_DICTIONARY_HEX.
  */
 static const char LOG_HEX_SEP[10] = "##ZLOGV1##";
 
 static const struct device *uart_dev;
+static struct k_sem sem;
+static volatile bool in_panic;
+static bool use_async;
 
-static int char_out(uint8_t *data, size_t length, void *ctx)
+static void uart_callback(const struct device *dev,
+			  struct uart_event *evt,
+			  void *user_data)
 {
-	ARG_UNUSED(ctx);
+	switch (evt->type) {
+	case UART_TX_DONE:
+		k_sem_give(&sem);
+		break;
+	default:
+		break;
+	}
+}
 
+static void dict_char_out(uint8_t *data, size_t length)
```

Comment:
Maybe rename this to `dict_char_out_hex()`? As this is only used for hexadecimal output.

Reply:
done